drivers/pppos.c

```diff
@@ -524,61 +534,19 @@ static void pppos_mainLoop(void* _state)
 
 static int pppos_netifUp(pppos_priv_t *state)
 {
-	char lcfg[256] = { 0 };
-	int line = 0;
-	FILE *fcfg = fopen(state->config_path, "r");
-	char *cfgval;
-	char *eq;
-
-	if (fcfg == NULL)
-		return 1;
-
-	if (state->apn[0]) {
-		fclose(fcfg);
-		return 0;
-	}
-
-	mutexLock(state->lock);
-	while (fgets(lcfg, sizeof(lcfg), fcfg) != NULL) {
-		line++;
-		if (lcfg[0] == '#')
-			continue;
-
-		if ((eq = strchr(lcfg, '=')) == NULL) {
-			log_error("[line %d] invalid format - missing '='", line);
-			continue;
-		}
-
-		lcfg[strcspn(lcfg, "\r\n")] = 0;
-
-		*eq = 0;
-		cfgval = eq + 1;
-
-		if (cfgval[0] == '"' || cfgval[0] == '\'') {
-			cfgval++;
-			cfgval[strlen(cfgval) - 1] = 0;
-		}
-
-		if (!strcasecmp(lcfg, "apn"))
-			strncpy(state->apn, cfgval, sizeof(state->apn) - 1);
-		else
-			log_warn("[line %d] unsupported option: %s (val: %s)", line, lcfg, cfgval);
-	}
-
 	condSignal(state->cond);
-	mutexUnlock(state->lock);
-	fclose(fcfg);
 
 	return 0;
 }
 
 
 static int pppos_netifDown(pppos_priv_t *state)
 {
-	if (state->apn[0]) {
-		state->apn[0] = 0;
-		pppapi_close(state->ppp, 0);
-	}
+	/* Unconditional use of pppapi_close() in the status callback
```

Comment:
good catch, I see some Kamil added this code. I suppose Your solution should break the connection more gracefully. 

This will work across the threads as conn_state is declared volatile. It can also be done with `want_connected` field I suggested above.